### PR TITLE
[REVIEWS-91] Create a custom eevnt to update RatingsSummary component

### PR DESCRIPTION
**Pull request by Arturo777**
#### What problem is this solving?

The rating summary component was not updating when a new review was saved in the front.
I create a new file inside the utils folder to export the eventBus object that contains three functions, i'm using the dispatch function to create a custom event, and the rest of the function could help for future implementations related with event listeners.

#### How to test it?
Go the workspace to test and create a new review, you should see the new review added in the front as well as the rating-summary block:
Screen Shot 2022-05-12 at 14 30 15

[Workspace to test](https://arturoreviews--sandboxusdev.myvtex.com/invicta-pro-diver-swiss-movement-quartz-watch---stainless-steel-case---model-0003/p)

### CODE REVIEW

File: react/RatingSummary.tsx [hunk 1]
```diff
@@ -196,9 +219,42 @@ function RatingSummary() {
...
+    averageRatingByProductId,
+    loadingAverageRatingByProductId,
+    refetchAverageRatingByProductId,
+  ])
+
+  useEffect(() => {
+    eventBus.on('reviewSaved', () => {
+      refetchTotalReviewsByProductId()
+      refetchAverageRatingByProductId()
+    })
```
Comment (arturmagalhaesjr) [inline react/RatingSummary.tsx:256]: you should put it outside of useEffect or use the unmount standard which uses the return on the useEffect to remove this listener

Comment (Arturo777) [reply react/RatingSummary.tsx:224]: ok, i also forgot to remove the old call, but now it's done. thank you.

File: react/RatingSummary.tsx [hunk 2]
```diff
@@ -218,7 +242,9 @@ function RatingSummary() {
         args: { authenticated: true },
       })
     })
-  }, [])
+
+    return () => eventBus.remove('reviewSaved')
+  }, [refetchTotalReviewsByProductId, refetchAverageRatingByProductId])
```
Comment (arturmagalhaesjr) [inline react/RatingSummary.tsx:247]: it's not going to work. 
you must isolate that as the example below:

```ts
useEffect (() => {
   const onReviewSaved = () => {
      refetchTotalReviewsByProductId()
      refetchAverageRatingByProductId()
   }
   eventBus.on('reviewSaved', onReviewSaved);

   ...

   return () => eventBus.remove('reviewSaved', onReviewSaved)
}, [refetchTotalReviewsByProductId, refetchAverageRatingByProductId]))
```
Even though you are calling the remove by passing null as a parameter, I suggest you it to ensure that function will remove without side effects.

**Review by arturmagalhaesjr — CHANGES_REQUESTED** [review]

**Review by arturmagalhaesjr — APPROVED** [review]
LGTM

**Pull request merged**